### DISCUSSION TITLE
A lot of timeouts during push/pull

### DISCUSSION THREAD

**Discussion by DimanNe** [community/community#64620] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

I started seeing a lot of issues like this:

At 12:42:58 everything is fine:
```
git push
Everything up-to-date
```

At 12:45:28:

```
git push -v
Pushing to github.com:DimanNe/....git
ssh: connect to host github.com port 22: Connection timed out
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

In other cases, if it is not timing out, it is VEEERY slow.

**Comment by arafajs** [comment]
This can happen for a few reasons:

- The GitHub server may be overloaded or down.
- Your computer may not be able to connect to the internet.
- There may be a firewall blocking the SSH connection.
- There may be a problem with your SSH configuration.

To troubleshoot the issue, you can try the following steps:

- Check the status of the GitHub server. You can do this by visiting the GitHub status page: https://status.github.com/.
- Check your internet connection. You can do this by running a speed test or trying to access a website that you know is up.
- Check your firewall settings. Make sure that the SSH port (22) is not blocked.
- Check your SSH configuration. Make sure that the hostname and port are correct.

In addition to the above, here are some other things you can try:

- Use a different SSH client.
- Try using a VPN.
- Try connecting to GitHub from a different network.
- Restart your computer and router.

**Reply by DimanNe** [reply]
It works perfectly fine after a while for first several pushes/pulls.

It looks like some sort of throttling. Unfortunately on https://status.github.com there is no information about throttling/rate-limiting that they apply...

**Reply by arafajs** [reply]
I think GitHub has a rate limit of requests per minute. If you are exceeding your rate limit, you will start to see errors.

**Reply by DimanNe** [reply]
Do you know what the numbers are?

I am not trying to do anything crazy: it is around 10-20 submodules that I am trying to pull/push...

**Comment by j-fulbright** [comment]
We have been seeing this issue for the first time over the last couple of weeks. It has never been an issue before.

**Comment by one1zero1one** [comment]
I have the same issue,  did not happen before. Especially frustrating when working with gitops repos where there are fast iterations.
To sove it, I switch on a VPN and then switch back, but I would love to know the limits of requests - or a better way to address it.

**Comment by zaphodbe** [comment]
This seems to be happening more and more.  Tried all the different hosts and ports with much the same thing happening.  Some day's it is good others it is practically impossible to access GitHub.  I don't see any response to any of this from the GitHub people directly.  There must be a better solution.
